src/ddscxx/src/org/eclipse/cyclonedds/sub/qos/DataReaderQosDelegate.cpp

```diff
@@ -210,7 +220,8 @@ DataReaderQosDelegate::operator==(const DataReaderQosDelegate& other) const
            other.resources_   == resources_   &&
            other.ownership_   == ownership_   &&
            other.tfilter_     == tfilter_     &&
-           other.lifecycle_   == lifecycle_;
+           other.lifecycle_   == lifecycle_   &&
+           other.datarepresentation_ == datarepresentation_;
```

Comment:
Please also encapsulate this line with the conditional OMG_DDS_EXTENSIBLE_AND_DYNAMIC_TOPIC_TYPE_SUPPORT macro. (BTW: I see that in this particular case the embedding OMG_DDS_OWNERSHIP_SUPPORT macro was also missing for the "ownership" policy).